### PR TITLE
Implement Saving Screen Templates Feature

### PR DESCRIPTION
**Pull request by sanjacornelius**
This PR implements both frontend and backend functionalities for saving a screen template. This feature allows users to select the 'Save as Template' option from the screen listing option menu. Upon selection, a new 'Create Template' modal appears, allowing the user to configure various settings such as name, description, version, thumbnails, category, and public settings for the template. Upon saving, the template is created, and a success message is displayed to the user.

## Solution
- Added and refactored existing frontend components and UI elements to faciliate template creation
- Implemented backend logic to handle saving screen templates with specificed configurations
- integrated frontend and backend components. 

## How to Test

**Test Case A**

1. Navigate to the Screen listing page.
2. Select the 'Save as Template' option from the options menu of a screen
3. Fill in the details in the 'Create Template' modal.
4. Click the 'Save' button
5. Verify that the template is created successfully
6. Confirm that a success message is displayed to indicate successful template creation.
7. Verify the created template in 'My Templates' or 'Public Templates' tab.

**Test Case B**

1. Navigate to the Screen listing page.
2. Select the 'Save as Template' option from the options menu of a screen
3. Fill in the details in the 'Create Template' modal.
4. Leave some required data empty
5. Verify the 'Save' button is disabled.
6. Add in missing data
7. Verify the 'Save' button is enabled
8. Select the 'Save' button
9. Confirm that a success message is displayed to indicate successful template creation.
10. Verify the created template in 'My Templates' or 'Public Templates' tab.

**Test Case C**

1. Navigate to the Screen listing page.
2. Select the 'Save as Template' option from the options menu of a screen
3. Fill in the details in the 'Create Template' modal.
4. Ensure the name of the template is the same as an existing template.
5. Select the 'Save' button
6. Ensure a Existing template warning appears
7. Verify that both 'Update' and 'Save' buttons work as expected

## Related Tickets & Packages
- [FOUR-13538](https://processmaker.atlassian.net/browse/FOUR-13538)

ci:next
ci:deploy

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

[FOUR-13538]: https://processmaker.atlassian.net/browse/FOUR-13538?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13348f392.engk8s.processmaker.net

File: ProcessMaker/Templates/ScreenTemplate.php [hunk 1]
```diff
@@ -268,25 +277,66 @@ protected function getRequestInclude(Request $request) : array
...
-            // offer to choose between “Update Template” and “Save as New Template”
-            return ['id' => $template->id, 'name' => $name];
+        return $screenTemplate;
+    }
+
+    private function updateScreenTemplateData(array $data, array $payload)
+    {
+        ScreenTemplates::where('name', $data['name'])->update([
+            'description' => $data['description'],
+            'is_public' => $data['make_public'] ? 1 : 0,
```
Comment (eiresendez) [inline ProcessMaker/Templates/ScreenTemplate.php:306]: This seems to be working fine; but when I created a template without clicking on the "Make public" checkbox, my template appeared in the Public Templates section tab... I believe it should only appear under "My Templates"

maybe, this issue could be addressed in a separate ticket?

File: database/migrations/2024_03_09_004053_update_unique_template_id_default_value_in_screen_templates.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('screen_templates', function (Blueprint $table) {
+            $table->string('unique_template_id')->nullable()->change();
```
Comment (eiresendez) [inline database/migrations/2024_03_09_004053_update_unique_template_id_default_value_in_screen_templates.php:14]: As an observation, I think that this value is not being used yet. It appears empty in my template, If not needed we can consider removing it

**Review by eiresendez — APPROVED** [review]
Functional test works great! 👍

File: database/migrations/2024_03_09_004053_update_unique_template_id_default_value_in_screen_templates.php (hunk 2 above)
Comment (sanjacornelius) [reply database/migrations/2024_03_09_004053_update_unique_template_id_default_value_in_screen_templates.php:14]: @eiresendez This value will be used by the Default Templates created by the PLG team

File: ProcessMaker/Templates/ScreenTemplate.php (hunk 1 above)
Comment (sanjacornelius) [reply ProcessMaker/Templates/ScreenTemplate.php:306]: @eiresendez Filtering of the My template and public templates will be implemented in another ticket

**Pull request merged**